### PR TITLE
Update search bar logic and style

### PR DESCRIPTION
**Pull request by hslee2008**
# What Changed
- [x] Make searchbar component and drawer go behind the drawer
- [x] Make width for light/dark the same
- [x] Add media to control the width of the input according to screen size
- [x] Align logo with the baseline of text
- [x] Add custom css for the whole searchbar style to customize it
- [x] Update deprecated code (useColorMode().isDarkTheme)
- [x] Make the dropdown appear fully in mobile (Also, hide `.docs-searchbar-suggestion--text` in order to see the texts better)
- [x] Change style of search component when theme is changed (immediately)

# Before and After Comparison

### Change style of search component when theme is changed (immediately)
This can be tested in [Latest Netlify Deployment](https://deploy-preview-600--tauri.netlify.app/).

### Make the dropdown appear fully in mobile
before mobile
after mobile

### Logo
before logo
after logo

### Mobile Drawer
before drawer
after drawer

### Center title so that it looks better in both mobile and computer sized screens
before center
after center

### @media (Other screen sizes as well, 1185, 995, 600)
before media
after media

### Sizes of width is same in both light/dark theme
before light
after light

### Deprecated code
<details>
  <summary>Log</summary>
  <p>
<pre>
colorMode.tsx?7cd1:130 `useColorMode().isDarkTheme` is deprecated. Please use `useColorMode().colorMode === "dark"` instead.
get isDarkTheme @ colorMode.tsx?7cd1:130
Component @ index.js?a860:1
renderWithHooks @ react-dom.development.js?ac89:16175
mountIndeterminateComponent @ react-dom.development.js?ac89:20913
beginWork @ react-dom.development.js?ac89:22416
beginWork$1 @ react-dom.development.js?ac89:27381
performUnitOfWork @ react-dom.development.js?ac89:26513
workLoopSync @ react-dom.development.js?ac89:26422
renderRootSync @ react-dom.development.js?ac89:26390
performSyncWorkOnRoot @ react-dom.development.js?ac89:26041
flushSyncCallbacks @ react-dom.development.js?ac89:12009
flushSync @ react-dom.development.js?ac89:26157
legacyCreateRootFromDOMContainer @ react-dom.development.js?ac89:29477
legacyRenderSubtreeIntoContainer @ react-dom.development.js?ac89:29503
render @ react-dom.development.js?ac89:29587
eval @ clientEntry.js?1755:23
Promise.then(asynchronous)
eval @ clientEntry.js?1755:23
./node_modules/@docusaurus/core/lib/client/clientEntry.js @ main.js:253
__webpack_require__ @ runtime~main.js:36
__webpack_exec__ @ main.js:2016
(Anonymous) @ main.js:2017
__webpack_require__.O @ runtime~main.js:83
(Anonymous) @ main.js:2018
webpackJsonpCallback @ runtime~main.js:1331
(Anonymous) @ main.js:9
react-dom.development.js?ac89:86 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

Please update the following components: RevealBase
printWarning @ react-dom.development.js?ac89:86
warn @ react-dom.development.js?ac89:49
ReactStrictModeWarnings.flushPendingUnsafeLifecycleWarnings @ react-dom.development.js?ac89:12203
flushRenderPhaseStrictModeWarningsInDEV @ react-dom.development.js?ac89:27263
commitRootImpl @ react-dom.development.js?ac89:26658
commitRoot @ react-dom.development.js?ac89:26638
performSyncWorkOnRoot @ react-dom.development.js?ac89:26073
flushSyncCallbacks @ react-dom.development.js?ac89:12009
flushSync @ react-dom.development.js?ac89:26157
legacyCreateRootFromDOMContainer @ react-dom.development.js?ac89:29477
legacyRenderSubtreeIntoContainer @ react-dom.development.js?ac89:29503
render @ react-dom.development.js?ac89:29587
eval @ clientEntry.js?1755:23
Promise.then(asynchronous)
eval @ clientEntry.js?1755:23
./node_modules/@docusaurus/core/lib/client/clientEntry.js @ main.js:253
__webpack_require__ @ runtime~main.js:36
__webpack_exec__ @ main.js:2016
(anonymous) @ main.js:2017
__webpack_require__.O @ runtime~main.js:83
(anonymous) @ main.js:2018
webpackJsonpCallback @ runtime~main.js:1331
(anonymous) @ main.js:9
index.js?a860:1 null
react_devtools_backend.js:4026 `useColorMode().isDarkTheme` is deprecated. Please use `useColorMode().colorMode === "dark"` instead.
    at Component (webpack-internal:///./src/theme/SearchBar/index.js:9:111)
    at div
    at NavbarSearch (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/Navbar/Search/index.js:13:36)
    at SearchNavbarItem (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/NavbarItem/SearchNavbarItem.js:14:40)
    at NavbarItem (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/NavbarItem/index.js:15:112)
    at NavbarItems (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/Navbar/Content/index.js:24:128)
    at div
    at div
    at NavbarContentLayout (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/Navbar/Content/index.js:24:625)
    at NavbarContent (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/Navbar/Content/index.js:24:1129)
    at nav
    at NavbarLayout (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/Navbar/Layout/index.js:19:422)
    at Navbar
    at NavbarSecondaryMenuDisplayProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/navbarSecondaryMenu/display.js:20:327)
    at NavbarMobileSidebarProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/navbarMobileSidebar.js:22:423)
    at NavbarSecondaryMenuContentProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/navbarSecondaryMenu/content.js:15:166)
    at NavbarProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/navbarUtils.js:21:38)
    at HtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:28:56)
    at PluginHtmlClassNameProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/metadataUtils.js:31:52)
    at DocsPreferredVersionContextProviderUnsafe (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/docsPreferredVersion.js:32:392)
    at DocsPreferredVersionContextProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/docsPreferredVersion.js:35:60)
    at ScrollControllerProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/utils/scrollUtils.js:19:498)
    at TabGroupChoiceProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/tabGroupChoice.js:15:1390)
    at AnnouncementBarProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/announcementBar.js:24:459)
    at ColorModeProvider (webpack-internal:///./node_modules/@docusaurus/theme-common/lib/contexts/colorMode.js:26:1255)
    at LayoutProviders (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/LayoutProviders/index.js:19:39)
    at Layout (webpack-internal:///./node_modules/@docusaurus/theme-classic/lib-next/theme/Layout/index.js:24:33)
    at Home (webpack-internal:///./src/pages/index.js:26:959)
    at RouteContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/routeContext.js:15:70)
    at LoadableComponent (webpack-internal:///./node_modules/react-loadable/lib/index.js:138:32)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:449:29)
    at Switch (webpack-internal:///./node_modules/react-router/esm/react-router.js:651:29)
    at Route (webpack-internal:///./node_modules/react-router/esm/react-router.js:449:29)
    at ClientLifecyclesDispatcher (webpack-internal:///./node_modules/@docusaurus/core/lib/client/ClientLifecyclesDispatcher.js:14:710)
    at PendingNavigation (webpack-internal:///./node_modules/@docusaurus/core/lib/client/PendingNavigation.js:16:112)
    at Root (webpack-internal:///./node_modules/@docusaurus/core/lib/client/theme-fallback/Root/index.js:19:25)
    at BrowserContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/browserContext.js:20:136)
    at DocusaurusContextProvider (webpack-internal:///./node_modules/@docusaurus/core/lib/client/docusaurusContext.js:20:505)
    at ErrorBoundary (webpack-internal:///./node_modules/@docusaurus/core/lib/client/exports/ErrorBoundary.js:14:108)
    at App (webpack-internal:///./node_modules/@docusaurus/core/lib/client/App.js:27:241)
    at Router (webpack-internal:///./node_modules/react-router/esm/react-router.js:80:30)
    at BrowserRouter (webpack-internal:///./node_modules/react-router-dom/esm/react-router-dom.js:58:35)
    at r (webpack-internal:///./node_modules/react-helmet-async/lib/index.module.js:17:7350)
overrideMethod @ react_devtools_backend.js:4026
get isDarkTheme @ colorMode.tsx?7cd1:130
Component @ index.js?a860:1
renderWithHooks @ react-dom.development.js?ac89:16175
updateFunctionComponent @ react-dom.development.js?ac89:20387
beginWork @ react-dom.development.js?ac89:22430
beginWork$1 @ react-dom.development.js?ac89:27381
performUnitOfWork @ react-dom.development.js?ac89:26513
workLoopSync @ react-dom.development.js?ac89:26422
renderRootSync @ react-dom.development.js?ac89:26390
performSyncWorkOnRoot @ react-dom.development.js?ac89:26041
flushSyncCallbacks @ react-dom.development.js?ac89:12009
flushSyncCallbacksOnlyInLegacyMode @ react-dom.development.js?ac89:11988
scheduleUpdateOnFiber @ react-dom.development.js?ac89:25438
dispatchSetState @ react-dom.development.js?ac89:17389
updateWindowSize @ useWindowSize.ts?9fc6:46
setTimeout(asynchronous)
eval @ useWindowSize.ts?9fc6:46
commitHookEffectListMount @ react-dom.development.js?ac89:23049
commitPassiveMountOnFiber @ react-dom.development.js?ac89:24816
commitPassiveMountEffects_complete @ react-dom.development.js?ac89:24781
commitPassiveMountEffects_begin @ react-dom.development.js?ac89:24768
commitPassiveMountEffects @ react-dom.development.js?ac89:24756
flushPassiveEffectsImpl @ react-dom.development.js?ac89:26990
flushPassiveEffects @ react-dom.development.js?ac89:26935
eval @ react-dom.development.js?ac89:26725
workLoop @ scheduler.development.js?bcd2:266
flushWork @ scheduler.development.js?bcd2:239
performWorkUntilDeadline @ scheduler.development.js?bcd2:533
</pre></p>
</details>

### CODE REVIEW

**Comment by lorenzolewis** [comment]
Really good so far, thank you very much! 

For the theme changes, I believe you'll want to add `isDarkTheme` to the array here: https://github.com/HyunseungLee-Travis/tauri-docs/blob/361a25c9ea23150d419466dc51de30dc035efd88/src/theme/SearchBar/index.js#L22

The issue that I ran into with that is that React changes the class name between renders (I think) and so that selector `.search-bar-input` somehow gets invalidated when the hook triggers a re-render.

Another minor details is that the search bar doesn't seem to fade with the rest of the content when the nav drawer is out on mobile, not sure if that would be an easy fix or not:

Screenshot 2022-05-14 at 10 01 45
Screenshot 2022-05-14 at 10 01 54


(Throwing in a reference to https://github.com/tauri-apps/tauri-docs/issues/372)

**Comment by hslee2008** [comment]
Changed the `z-index` of backdrop to be the same as the input to make the input hide behind the backdrop.

**Comment by hslee2008** [comment]
But I am not exactly yet certain but putting `isDarkTheme` as a dependant inside the `useEffect` array makes the app crash.

crash

Still investigating.

**Comment by lorenzolewis** [comment]
> But I am not exactly yet certain but putting `isDarkTheme` as a dependant inside the `useEffect` array makes the app crash.
> 
> crash
> 
> Still investigating.

Yeah, this was the same issue I was running into. It isn't *that* big of a deal at all leaving it as-is. It will catch it on the next page load anyway

**Comment by lorenzolewis** [comment]
Maybe a bit of weird text sizing going on when it's a really small device here?

Screenshot 2022-05-14 at 11 35 43
Screenshot 2022-05-14 at 11 35 49

File: src/theme/SearchBar/searchbar.css [hunk 1]
```diff
@@ -0,0 +1,91 @@
+.searchbox {
```
Comment (lorenzolewis) [inline src/theme/SearchBar/searchbar.css]: Let's rename this file to `styles.css` to it aligns a bit closer to other components (even though they're able to use modules)

Comment (lorenzolewis) [reply src/theme/SearchBar/searchbar.css]: And then in this file if you could just add comments above the selectors noting what that specific styling does then that will help a future contributor a lot

**Review by lorenzolewis — CHANGES_REQUESTED** [review]
Overall very very nice, just a few semantic and comment things and then it should be good!

**Review by lorenzolewis — APPROVED** [review]
Perfect, thanks a million for contributing! Feel free to join over in discord so I can give you a shout!

**Pull request merged**